File: зад2.cs
using System;

class MainClass {
  static void Main() {
    Console.WriteLine("Введите строку:  ");
    string rings = Console.ReadLine();
	//Проверка
	for (int i = 0; i < rings.Length ; i += 2) {
		if (rings[i] == 'R' || rings[i] == 'W' || rings[i] == 'B') {
			if (rings[i + 1] < '0' || rings[i + 1]>'9') {
				Console.WriteLine("Значение было введено неверно");
			}
		}else {
			Console.WriteLine("Значение было введено неверно");
		}
	}
	
int[] sticks = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
	//Считаем кол-во колец на каждом стержне
for (int i = 0; i < rings.Length ; i += 2) {
	
		if (rings[i] == 'R') {
			sticks[rings[i + 1] - '0'] *= 2;
		}
		else if (rings[i] == 'W') {
			sticks[rings[i + 1] - '0'] *= 3;
		}
		else if (rings[i] == 'B') {
			sticks[rings[i + 1] - '0'] *= 5;
		}
	}
	int counter = 0;
	//На скольких кольцах стержни одного цвета
	for (int i=0; i<sticks.Length;i++) {
		if (sticks[i] % 30 == 0) {
			counter++;
		}
	}
	Console.WriteLine(counter);
  }
}

File: зад2.rs
use std::io;

fn main()  {
//Ввод строки
    let mut rings = String::new();
    println!("Введите строку:  ");
    io::stdin().read_line(&mut rings).expect("Failed to read input");
    println!("Ваша строка: {}", rings);
    let length = rings.chars().count();
    //Проверка
for i in 0..length-2{
if i%2 ==0 {
   let mut j=i+1;
    //Перевод строки по индексу сначала в u8 , а потом в чар для сравнения
    let b1: u8 = rings.as_bytes()[j];
    let c1: char = b1 as char;
    let b: u8 = rings.as_bytes()[i];
    let c: char = b as char;
	if c =='R' || c =='B' || c =='W'{
		if c1 < '0' || c1>'9'{
		println!("Значение было введено неверно");
		}
	}else {
		println!("Значение было введено неверно");
	}
	}
}
//Поиск
 let mut sticks: [i32; 10] = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
  for i in 0..length-2{
  if i%2 == 0 {
  let mut j=i+1;
  let b1: u8 = rings.as_bytes()[j];
  let c1: char = b1 as char;
  let b: u8 = rings.as_bytes()[i];
  let c: char = b as char;
  //let mut z = b1.trim().parse::<i64>().unwrap();
  if c == 'R'{
		sticks[(c1 as usize)-48] *= 2;
	}else if c == 'W'{ 
		sticks[(c1 as usize)-48] *= 3;
	}else if c == 'B'{
		sticks[(c1 as usize)-48] *= 5;
	}
    }
  }
 let mut counter = 0;
 //На скольких кольцах стержни одного цвета
 for i in 0..sticks.len()-1 {
  if sticks[i] % 30 == 0 {
   counter += 1;
  }
 }
 println!("{}", counter);
  
}
